dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -1477,6 +1475,29 @@ BlobStatPtr BlobStore::BlobStats::blobIdToStat(BlobFileId file_id, bool ignore_n
     return nullptr;
 }
 
+String BlobStore::BlobStats::blobIdToPath(BlobFileId file_id)
+{
+    auto guard = lock();
+    return blobIdToPathImpl(file_id, guard);
+}
+
+String BlobStore::BlobStats::blobIdToPathImpl(BlobFileId file_id, const std::lock_guard<std::mutex> &)
+{
+    for (const auto & [path, stats] : stats_map)
+    {
+        for (const auto & stat : stats)
+        {
+            if (stat->id == file_id)
+            {
+                return path;
+            }
+        }
+    }
+    throw Exception(fmt::format("Can't find BlobStat with [blob_id={}]",
+                                file_id),
+                    ErrorCodes::LOGICAL_ERROR);
+}
```

Comment:
In BlobStore::removePosFromStats, we remove the stat from `blob_stats` first, after that call `getBlobFile` -> `getBlobFileParentPath` -> `blobIdToPath`. Exception will be thrown under this case?

https://github.com/pingcap/tiflash/blob/a2d7f16b24e0e5434c9c994be51d7542041512b0/dbms/src/Storages/Page/V3/BlobStore.cpp#L506-L515

Reply:
Yes, we need remove BlobFile first.